feat: add support for FoxLake

jairuigou opened the pull request:
Notes:
1. Since foxlake needs to use the helm chart hooks to automatically create some resources, and also needs to create serviceaccount and clusterrolebinding, it needs to be installed through `helm install` temporarily until kbcli supports.
2. Since foxlake-cluster uses `serviceAccountName` in cluster componentSpec, the KubeBlocks version must be at least `v0.5.0-beta.15` or higher.
3. The latest KEDA requires Kubernetes cluster version 1.24 and higher, so temporarily cannot use `kbcli playground init` to create a cluster.

Usage:
```
# install kbcli and kubeblocks
curl -fsSL https://kubeblocks.io/installer/install_cli.sh |bash -s 0.5.0-beta.27
kbcli kubeblocks install

cd kubeblocks
make kbcli
# install foxlake cd,cv
helm install foxlake deploy/foxlake
# install foxlake-cluster
helm install foxlake-cluster deploy/foxlake-cluster
# show connect example to get the random root password
./bin/kbcli cluster connect foxlake-cluster --show-example
# connect via kbcli
./bin/kbcli cluster connect foxlake-cluster
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/apecloud/kubeblocks?pullRequest=3048) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/apecloud/kubeblocks?pullRequest=3048) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/apecloud/kubeblocks?pullRequest=3048) it.</sub>

jairuigou commented:
I have updated the Helm chart with `KubeBlocks v0.6.0-beta.6`. The latest version of FoxLake now supports creating default storage during the server initialization, so I removed the Helm hook. And after discussion, we have concluded that users should manage their own storage service, whether it's MinIO or AWS S3, so I removed the minio component. Users can now use the `--set` flag to pass default storage configurations.

Install example:
```
# install foxlake cd,cv
helm install foxlake deploy/foxlake
# install foxlake cluster
helm install foxlake-cluster deploy/foxlake-cluster --set defaultStorage.create=true,defaultStorage.accessKey="admin",defaultStorage.secretKey="F0E2ZnIPz4",defaultStorage.endpoint="minio://minio:9000/foxlake/default"
```

jairuigou commented on deploy/foxlake-cluster/templates/clusterrolebinding.yaml:
```diff
@@ -0,0 +1,14 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
```
ClusterRoleBinding is necessary because foxlake needs to be granted cluster-wide access permissions. I have tested with RoleBinding, the `create cluster` function can not work properly

nashtsai commented on deploy/foxlake/templates/keda-addon.yaml:
```diff
@@ -0,0 +1,16 @@
+    chartLocationURL: https://kedacore.github.io/charts/keda-2.10.2.tgz
+
+  installable:
+    autoInstall: true
```
having autoInstall=true, woud be better to have KEDA as sub-chart instead of addon, ref: https://helm.sh/docs/chart_template_guide/subcharts_and_globals/

jairuigou replied:
> having autoInstall=true, woud be better to have KEDA as sub-chart instead of addon, ref: https://helm.sh/docs/chart_template_guide/subcharts_and_globals/

Made this change in the latest commit.
